x-pack/elastic-agent/pkg/agent/application/pipeline/dispatcher/dispatcher.go

```diff
@@ -74,7 +76,15 @@ func (ad *ActionDispatcher) key(a fleetapi.Action) string {
 }
 
 // Dispatch dispatches an action using pre-registered set of handlers.
-func (ad *ActionDispatcher) Dispatch(acker store.FleetAcker, actions ...fleetapi.Action) error {
+func (ad *ActionDispatcher) Dispatch(ctx context.Context, acker store.FleetAcker, actions ...fleetapi.Action) (err error) {
+	span, ctx := apm.StartSpan(ctx, "dispatch", "app.internal")
+	defer func() {
+		if err != nil {
+			apm.CaptureError(ctx, err).Send()
+		}
```

Comment:
[Suggestion]
It's safe to call `apm.CaptureError` with a nil error and then call `Send`. Therefore it can be simplified.
```suggestion
		apm.CaptureError(ctx, err).Send()
```